src/goto-programs/goto_contractor.cpp

```diff
@@ -171,35 +169,89 @@ void goto_contractort::contractor()
   log_status("{}", oss.str());
 }
 
-ibex::CmpOp goto_contractort::get_complement(ibex::CmpOp op)
+ibex::Ctc *goto_contractort::create_contractor_from_expr2t(const expr2tc &expr)
 {
-  switch(op)
+  ibex::Ctc *contractor = nullptr;
+  auto base_object = get_base_object(expr);
+
+  if(is_unsupported_operator_in_contractor(expr))
   {
-  case ibex::GEQ:
-    return ibex::LT;
-  case ibex::GT:
-    return ibex::LEQ;
-  case ibex::LEQ:
-    return ibex::GT;
-  case ibex::LT:
-    return ibex::GEQ;
-  default:
-    log_status("cant process equal");
-    break;
+    auto cons = create_constraint_from_expr2t(base_object);
```

Comment:
is it's unsupported, shouldn't we just return nullptr here?

Reply:
I changed the name of the function to show its purpose. Because if we encounter these operators we will create constraints, not contractors.